bin/read_async.c

```diff
@@ -51,14 +59,30 @@ int main(int argc, char *argv[]){
     return 0;
 }
 
+static int setup_context(unsigned queue_depth, struct io_uring *ring) {
+  int ret;
+  ret = io_uring_queue_init(queue_depth, ring, 0);
+  if(ret < 0) {
+    fprintf(stderr, "queue_init: %s\n", strerror(-ret));
+    return -1;
+  }
+
+  return 0;
+}
+
 int prep_read_request(char *filename, int id, struct io_uring *ring){
+    off_t sz;
     int fd = open(filename, O_RDONLY);
     if (fd < 0){
         perror("open");
         return 1;
     };
-    off_t sz = get_file_size(fd);
+    if(get_file_size(fd, &sz)) {
+      return 1;
+    }
     struct io_uring_sqe *sqe = io_uring_get_sqe(ring);
+    assert(sqe);
```

Comment:
The error handling for `sqe` is different from the rest? Why not
```suggestion
    if !(sqe) {
        return 1;
    }
```
or isit standard practice when you're using library functions that don't give you a choice?